src/annotator/bucket-bar.js

```diff
@@ -1,24 +1,30 @@
-import Delegator from '../delegator';
-
 import { render } from 'preact';
-import Buckets from '../components/Buckets';
+import Buckets from './components/Buckets';
+
+import { anchorBuckets } from './util/buckets';
 
-import { anchorBuckets } from '../util/buckets';
+/**
+ * @typedef BucketBarOptions
+ * @prop {string[]} scrollables  // Selectors for the scrollable elements on the page
+ */
 
-export default class BucketBar extends Delegator {
-  constructor(container, options, annotator) {
+export default class BucketBar {
+  /**
+   * @param {HTMLElement} container
+   * @param {BucketBarOptions} options
+   * @param {Pick<import('./guest').default, 'anchors'|'selectAnnotations'>} guest
+   */
+  constructor(container, options, guest) {
     const defaultOptions = {
-      // Selectors for the scrollable elements on the page
       scrollables: [],
     };
 
-    const opts = { ...defaultOptions, ...options };
+    this.options = { ...defaultOptions, ...options };
```

Comment:
There is no real need to have `this.options` any more. You could just have a `this.scrollables` argument which is initialized to an empty array if there is no `scrollables` property in `options`.

Reply:
I addressed this as you suggested.